keyboards/skyloong/gk68/pro/keymaps/via/keymap.c

```diff
@@ -0,0 +1,172 @@
+    ),
+
+    [_WIN_INDEX] =  LAYOUT_all(
+         KC_GRV,    KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,       KC_F6,    KC_F7,      KC_F8,    KC_F9,          KC_F10,      KC_F11,   KC_F12,    KC_DEL,   KC_MUTE,
+        _______,  _______,    TO(0),    TO(1),  _______,  _______,     _______,  _______,    _______,  _______,         _______,    RGB_RMOD,  RGB_MOD,   RGB_TOG,   KC_MPLY,
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,  _______,    _______,  _______,         _______,     _______,             _______,   KC_HOME,
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,  _______,    _______,  _______,         _______,               _______,   RGB_VAI,    KC_END,
+        _______,  _______,  _______,            _______,               _______,              _______,  KC_RALT,         _______,      KC_APP,  RGB_SPD,   RGB_VAD,   RGB_SPI
+    ),
+
+    [_MAC_INDEX] = LAYOUT_all(
+        KC_GRV ,   KC_F14,   KC_F15, C(KC_UP),  G(KC_D), G(KC_SPC),  LSG(KC_4),  KC_MPRV,    KC_MPLY,  KC_MNXT,         KC_MUTE,     KC_VOLD,  KC_VOLU,    KC_DEL,   KC_MUTE,
+        _______,  _______,    TO(0),    TO(1),  _______,  _______,     _______,  _______,    _______,  _______,         _______,    RGB_RMOD,  RGB_MOD,   RGB_TOG,   KC_MPLY,
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,  _______,    _______,  _______,         _______,     _______,             _______,   KC_HOME,
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,  _______,    _______,  _______,         _______,               _______,   RGB_VAI,    KC_END,
+        _______,  KC_LALT,  KC_LGUI,            _______,               _______,              _______,  KC_RGUI,         _______,     KC_RALT,  RGB_SPD,   RGB_VAD,   RGB_SPI
+    )
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [_WIN] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(RGB_VAD, RGB_VAI)},
+    [_MAC] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(RGB_VAD, RGB_VAI)},
+    [_WIN_INDEX] = { ENCODER_CCW_CW(KC_MPRV, KC_MNXT), ENCODER_CCW_CW(RGB_HUD, RGB_HUI)},
+    [_MAC_INDEX] = { ENCODER_CCW_CW(KC_MPRV, KC_MNXT), ENCODER_CCW_CW(RGB_HUD, RGB_HUI)}
+};
+#endif
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+#    ifdef RGB_MATRIX_ENABLE
+        case RGB_TOG:
+            if (record->event.pressed) {
+                switch (rgb_matrix_get_flags()) {
+                    case LED_FLAG_ALL: {
+                        rgb_matrix_set_flags(LED_FLAG_NONE);
+                        rgb_matrix_set_color_all(0, 0, 0);
+                    } break;
+                    default: {
+                        rgb_matrix_set_flags(LED_FLAG_ALL);
+                    } break;
+                }
+            }
+         return false;
+#    endif
+     case TO(0):
```

Comment:
Not a fan of this.  DF would be a better option for this, I think. And worst case, add a `layer_clear` with it, to clear all the active layers.

Reply:
If I use the TO(), I can immediately indicate the layer change when I press TO(), but when I use the DF(), I need to release the DF() to indicate the layer change. From the perspective of user experience, the TO() is better than the DF(). @drashna